web/src/main/java/org/cbioportal/web/util/ClinicalDataEnrichmentUtil.java

```diff
@@ -0,0 +1,264 @@
+                                    clinicalDataCountItem -> clinicalDataCountItem));
+                }).collect(Collectors.toList());
+
+        attributes.forEach(clinicalAttribute -> {
+
+            String attributeKey = getClinicalAttributeKey(clinicalAttribute);
+
+            // get counts for all categories in all group for a given attribute
+            List<Map<String, Integer>> categoryCountsByGroup = dataCountsByGroupAndByAttribute.stream().map(e -> {
+                if (e.containsKey(attributeKey)) {
+                    return e.get(attributeKey).getCounts().stream()
+                            .collect(Collectors.toMap(ClinicalDataCount::getValue, ClinicalDataCount::getCount));
+                }
+                return new HashMap<String, Integer>();
+            }).collect(Collectors.toList());
+
+            // TODO: should we exclude NA category
+            Set<String> allPossibleCategories = categoryCountsByGroup.stream().flatMap(x -> x.keySet().stream())
+                    .collect(Collectors.toSet());
+
+            if (allPossibleCategories.size() > 1) {
+                DataTable2D dataTable = new DataTable2D();
+                int groupIndex = 0;
+                for (Map<String, Integer> groupCategoryCounts : categoryCountsByGroup) {
+                    Map<String, Integer> allCategoryCounts = allPossibleCategories.stream()
+                            .collect(Collectors.toMap(category -> category, category -> {
+                                return groupCategoryCounts.containsKey(category) ? groupCategoryCounts.get(category)
+                                        : 0;
+                            }));
+                    AssociativeArray categoryCounts = new AssociativeArray();
+                    categoryCounts.putAll(allCategoryCounts);
+                    dataTable.put(groupIndex++, categoryCounts);
+                }
+
+                double pValue = Chisquare.getPvalue(dataTable);
+
+                if (!Double.isNaN(pValue)) {
+                    ClinicalDataEnrichment clinicalEnrichment = new ClinicalDataEnrichment();
+                    clinicalEnrichment.setClinicalAttribute(clinicalAttribute);
+                    clinicalEnrichment.setpValue(BigDecimal.valueOf(pValue));
+                    clinicalEnrichment.setScore(BigDecimal.valueOf((Double) Chisquare.getScoreValue(dataTable)));
+                    clinicalEnrichment.setMethod("Chi-squared Test");
+                    clinicalEnrichments.add(clinicalEnrichment);
+                }
+
+            }
+        });
+
+        return clinicalEnrichments;
+    }
+
+    /**
+     * get data for all NUMBER datatype attributes for given samples
+     * 
+     * @param attributes
+     * @param samples
+     * @return
+     */
+    private Map<String, List<Double>> getNumericClinicalData(List<ClinicalAttribute> attributes, List<Sample> samples) {
+
+        List<String> _studyIds = new ArrayList<>();
```

Comment:
unlike JavaScript, variable names starting with `_` is not really conventional for Java.